### ISSUE TITLE
Feature Request: Allow destroy parameters to be optional

### ISSUE THREAD

**Issue by lucymhdavies** [devops-rob/terraform-provider-terracurl#11]
I've got a weird use-case, where I want to trigger a curl to create a thing, but I don't need to destroy it when doing a Terraform Destroy.

My workaround for now is:

```
  destroy_url            = "http://example.com"
  destroy_method         = "DELETE"
  destroy_response_codes = [405]
```

But if we could either make those parameters optional, that could be helpful.

**Comment by devops-rob** [comment]
Thanks for opening this issue @lucymhdavies . I can see how this would make it a bit more flexible for edge cases and happy to implement. Just double checking, does the data source not fit your use case?

**Comment by lucymhdavies** [comment]
It might do. I can POST with the Data Source, and still use a `depends_on` to control when TF makes that call.

Though I think a Data Source would run every time I do a plan/apply, whereas I only want something to be run once.

I'd have to experiment to see what happens.


For clarity, my use-case is that once I've created a bunch of workspaces, I want to trigger a plan on them.

(If I needed to wait until that Plan had completed, or needed the corresponding Destroy, I'd just use https://github.com/mitchellh/terraform-provider-multispace. But in this case, I just want to kick off an initial plan and call it good)

```
# Now that creds have been pushed, we can trigger a run
resource "terracurl_request" "trigger_workspaces" {
  name     = "trigger_workspaces"
  for_each = local.workspace_names

  depends_on = [
    aws_iam_role.aws-doormat-creds,
    module.azure-creds,
    tfe_workspace_run_task.tasks,
  ]

  url          = "https://app.terraform.io/api/v2/runs"
  method       = "POST"
  request_body = <<EOF
{
	"data": {
		"attributes": {
			"message": "Triggered by Terraform"
		},
		"type":"runs",
		"relationships": {
			"workspace": {
				"data": {
					"type": "workspaces",
					"id": "${tfe_workspace.workspace[each.key].id}"
				}
			}
		}
	}
}
EOF

  headers = {
    Authorization = "Bearer ${local.tfc_token}"
    Content-Type  = "application/vnd.api+json"
  }
  response_codes = [201]



  # TODO: remove these, after https://github.com/devops-rob/terraform-provider-terracurl/issues/11
  destroy_url            = "http://example.com"
  destroy_method         = "DELETE"
  destroy_response_codes = [405]
}

```

**Comment by lucymhdavies** [comment]
Switching out for `data "terracurl_request" "trigger_workspaces" {`

1st plan + apply creates the Run, as expected...

<img width="1249" alt="image" src="https://user-images.githubusercontent.com/1116529/183638615-9ba51014-be62-4024-8b55-33f4ef496e3b.png">

But yes, as I suspected, a subsequent plan + apply creates another Run, which is not what I want in this case

<img width="1245" alt="image" src="https://user-images.githubusercontent.com/1116529/183638737-6e5cc5d9-4021-4bb3-b8f3-f92c4051015e.png">

**Comment by devops-rob** [comment]
Makes sense. I'll build an updated version and link it to this issue

**Comment by devops-rob** [comment]
Hey @lucymhdavies - Thanks again for this feature request. I've implemented this in the v0.1.1 release which is now live in the Terraform Registry. Please do let mw know if you run into any issues.

**Comment by lucymhdavies** [comment]
Only just got around to updating. works great, no issues. Thanks :)
